src/string.js

```diff
@@ -600,7 +600,7 @@ function h$toHsStringMU8(arr, cc) {
 #else
 function h$toHsStringMU8(arr) {
 #endif
-    var accept = false, b, n = 0, cp = 0, r = HS_NIL;
+    var i = 0, accept = false, b, n = 0, cp = 0, r = HS_NIL;
```

Comment:
I'm pretty sure you meant:

    var i = arr.length - 1;

Reply:
Ah, whoops - fixing that now.